__tests__/integration/melt-tokens.test.js

```diff
@@ -209,18 +209,18 @@ describe('melt tokens', () => {
       .send({
         token: tokenA.uid,
         amount: 100,
-        deposit_address: await wallet1.getAddressAt(5),
+        deposit_address: await wallet1.getAddressAt(6),
```

Comment:
Why did you change it?

Reply:
I was tracking the changes that made this happen, the key change was that on the old lib we checked the address index with [`maxIndex > lastUsedIndex`](https://github.com/HathorNetwork/hathor-wallet-lib/blob/793bef622269e4462f62f85a35a7046a2b24b282/src/wallet.js#L2347-L2348) so if the max index of the last tx spent is the same as the last used address, the "current address" will not be updated and will point to an used address.

The new lib checks [`lastUsedAddressIndex <= maxIndexUsed`](https://github.com/HathorNetwork/hathor-wallet-lib/blob/aba5242e7bb137a390e97afc601e08ca815b3e39/src/utils/storage.ts#L264-L265) which will update the "current address" so it is always unused.

This difference is why here we choose the change address as the address 4 (which is used) and the current lib chooses the address 5 which is unused. Since the address 5 has another change utxo the `expects` below will fail, that's why I bumped the addresses, but we can also explicitly use the address 4 as change to the same effect, what do you think?

Reply:
This test was ensuring the wrong behavior since the expected change address was an address with a transaction already.
This is explained in the comment at the start of the test